File: client-app/src/app/models/order.ts
import { IContact, ContactFormValues } from "./contact";

export interface IOrder {
  id: string;
  orderNumber: number;
  clientId: string;
  client?: IContact;
  clientName?: string;
  type: boolean;
  closed: boolean;
  product: string;
  amount: ConstrainDouble;
  price: ConstrainDouble;
  dateOrderOpened: Date;
  dateOrderClosed: Date;
  notes: string;
}
export interface IOrderForm extends Partial<IOrder> {
  timeOrderOpened?: Date;
  timeOrderClosed?: Date;
}

export class OrderFormValues implements Partial<IOrder> {
  id: string = '';
  orderNumber: number = 0;
  clientId: string = '';
  // client: IContact = new ContactFormValues();
  clientName?: string = '';
  type: boolean = false;
  closed: boolean = false;
  product: string = '';
  amount: ConstrainDouble = 0;
  price: ConstrainDouble = 0;
  dateOrderOpened: Date = new Date();
  dateOrderClosed: Date = new Date(0);
  notes: string = '';

  constructor(init?: IOrder) {
    // if (init && init.dateOrderOpened) {
    //   init.timeOrderOpened = init.dateOrderOpened;
    // }
    Object.assign(this, init);
  }
}

File: Domain/Order.cs
using System;

namespace Domain
{
    public class Order
    {
        public Guid Id { get; set; }
        public int OrderNumber { get; set; }
        public Guid ClientId { get; set; }
        public virtual Contact Client { get; set; }
        public Boolean Type { get; set; }
        public Boolean Closed { get; set; }
        public string Product { get; set; }
        public Double? Amount { get; set; }
        public Double? Price { get; set; }
        public DateTime? DateOrderOpened { get; set; }
        public DateTime? DateOrderClosed { get; set; }
        public string Notes { get; set; }
    }
}
